code/game/machinery/turnstile.dm

```diff
@@ -0,0 +1,87 @@
+/obj/machinery/turnstile
+	name = "turnstile"
+	desc = "A mechanical door that permits one-way access and prevents tailgating."
+	icon = 'icons/obj/turnstile.dmi'
+	icon_state = "turnstile_map"
+	power_channel = ENVIRON
+	density = FALSE
+	armor = list(melee = 50, bullet = 50, laser = 50, energy = 50, bomb = 10, bio = 100, rad = 100, fire = 90, acid = 70)
+	anchored = TRUE
+	use_power = FALSE
+	idle_power_usage = 2
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF
+	layer = OPEN_DOOR_LAYER
+
+/obj/machinery/turnstile/Initialize()
+	. = ..()
+	icon_state = "turnstile"
+
+/obj/machinery/turnstile/CanAtmosPass(turf/T)
+	return TRUE // Let the air through
+
+/obj/machinery/turnstile/bullet_act(obj/item/projectile/P, def_zone)
+	return -1 // Let those bullets trough, dont ask why this is -1
+
+// Allow access for people being pulled by someone with access
+/obj/machinery/turnstile/proc/allowed_access(var/mob/B)
+	if(B.pulledby && ismob(B.pulledby))

```

Comment:
pulling people is ok, but grabbing them is not?

Reply:
Is there a way to check for grabs easily without even more snowflake code

Reply:
nope.

Reply:
WElp